pkg/util/validation/limits.go

```diff
@@ -265,8 +269,59 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.AlertmanagerMaxAlertsSizeBytes, "alertmanager.max-alerts-size-bytes", 0, "Maximum total size of alerts that a single tenant can have, alert size is the sum of the bytes of its labels, annotations and generatorURL. Inserting more alerts will fail with a log message and metric increment. 0 = no limit.")
 }
 
+// extensionsType is the type of the registered extensions.
+var extensionsType reflect.Type
+
+// newExtensionsOnlyLimitsConfig returns an interface{} value of a pointer to a struct of type:
+//
+//	struct {
+//	    Extensions T `yaml:"extensions"`
+//	    Throwaway map[string]interface{} `yaml:",inline"`
+//	}
+//
+// Where T is the registered extensionsType.
+// The second return ar
+func newExtensionsOnlyLimitsConfig() (any interface{}, getter func() interface{}) {
+	typ := reflect.StructOf([]reflect.StructField{
+		{
+			Name: "Extensions",
+			Type: extensionsType,
+			Tag:  `yaml:"extensions"`,
+		},
+		{
+			Name: "Throwaway",
+			Type: reflect.TypeOf(map[string]interface{}{}),
+			Tag:  `yaml:",inline"`,
+		},
+	})
+	newExtensions := reflect.New(typ).Interface()
+	return newExtensions, func() interface{} {
+		return reflect.ValueOf(newExtensions).Elem().Field(0).Interface()
+	}
+}
+
+// RegisterExtensions registers the extensions type and returns a function to get the extensions from a *Limits instance.
+func RegisterExtensions[E any]() func(*Limits) E {
+	extensionsType = reflect.TypeOf(new(E)).Elem()
```

Comment:
I'm confused by call to `Elem()` -- its documentation says:

> It panics if the type's Kind is not Array, Chan, Map, Pointer, or Slice.

But we would like to pass `struct` as a type, right?

Reply:
We're passing a pointer created by `new()`.